python/ray/tests/conftest.py

```diff
@@ -397,3 +402,70 @@ def unstable_spilling_config(request, tmp_path):
     ])
 def slow_spilling_config(request, tmp_path):
     yield create_object_spilling_config(request, tmp_path)
+
+
+@pytest.fixture
+def ray_start_chaos_cluster(request):
+    """Returns the cluster and chaos thread.
+
+    Run chaos_thread.start() to start the chaos testing.
+    NOTE: `cluster` is not thread-safe. `cluster`
+    shouldn't be modified by other thread once
+    chaos_thread.start() is called.
+    """
+    os.environ["RAY_num_heartbeats_timeout"] = "5"
+    os.environ["RAY_raylet_heartbeat_period_milliseconds"] = "100"
+    param = getattr(request, "param", {})
+    kill_interval = param.get("kill_interval", 2)
+    # Config of workers that are re-started.
+    head_resources = param["head_resources"]
+    worker_node_types = param["worker_node_types"]
+    timeout = param["timeout"]
+
+    # Use the shutdown RPC instead of signals because we can't
+    # raise a signal in a non-main thread.
+    def kill_raylet(ip, port, graceful=True):
+        raylet_address = f"{ip}:{port}"
+        channel = grpc.insecure_channel(raylet_address)
+        stub = node_manager_pb2_grpc.NodeManagerServiceStub(channel)
+        print(f"Sending a shutdown request to {ip}:{port}")
+        stub.ShutdownRaylet(
```

Comment:
can we verify the raylet is indeed being killed?

Reply:
I will add assert (it is only manually verified)